test/parallel/test-http2-client-socket-operations.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const h2 = require('http2');
+
+// test client.socket.destroy()
```

Comment:
I don't think much is gained by moving this here instead of having it in a separate file. The server still has to be instantiated separately and the two test cases aren't really connected. We also lose the history of the other file. I would strongly prefer to have the other test restored and have this taken out.

Reply:
Restored the code in `test-http2-client-socket-destroy.js` in new commit